src/main/java/com/example/petbuddybackend/repository/chat/ChatRoomRepository.java

```diff
@@ -71,4 +71,24 @@ SELECT MAX(cm3.createdAt)
         )
         """)
     Page<ChatRoomDTO> findByCaretakerEmailSortByLastMessageDesc(String email, Pageable pageable);
+
+    @Query("""
+        SELECT COUNT(cr)
+        FROM ChatRoom cr
+        WHERE (cr.client.email = :userEmail OR cr.caretaker.email = :userEmail)
+        AND EXISTS(
+            SELECT cm
+            FROM ChatMessage cm
+            WHERE cm.chatRoom.id = cr.id
+                AND cm.createdAt = (
+                    SELECT MAX(cm2.createdAt)
+                    FROM ChatMessage cm2
+                    WHERE cm2.chatRoom.id = cr.id
+                )
+                AND cm.sender.email != :userEmail
```

Comment:
To taka sytuacja:

Jest klient i chce wywołać to query. Ostatnia wiadomość została wysłana przez opiekuna, więc `createdAt` weźmie stamtąd. Jednocześnie klient nie będzie senderem więc tutaj nie zliczy tego jako 0?

Reply:
Jeśli byłoby tak jak mówię wcześniej, to też można uprościć to query. Chat działa tak, że jak już updateuje to `seenByRecipient` to dla każdej wiadomości, więc można uznać że `chatRoom` ma nieprzeczytaną wiadomość jeśli co najmniej jedna ma `sender != userEmail` i `seenByRecipient = false`

Reply:
Można by też pomyśleć o jakimś teście integracyjnym bo query jest dość skomplikowane, ale to jak uważasz

Reply:
wlasnie zastanawialem sie czy brac ostatnia wiadomosc, czy ogolnie jakakolwiek. A test dodalem dla query

Reply:
moglbym dodac 1 warunek w srodku tego zeby max czas wzielo od wiadomosci tylko odpowiedniego usera:
`AND cm.createdAt = (
                    SELECT MAX(cm2.createdAt)
                    FROM ChatMessage cm2
                    WHERE cm2.chatRoom.id = cr.id
                )`

Ale w sumie moge zobaczyc czy jakakolwiek jest nieodczytana